<a href="https://colab.research.google.com/github/mehkzhra/FlyRank-ML-Internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I have selected **Lane 2: Refresh / Content Opportunity Scoring**. My research question is: **Which content pages should be reviewed first for refresh, expansion, protection, pruning, or monitoring?**
I selected this lane because the dataset contains many pages with different levels of visibility, freshness, engagement, and performance movement.
A ranked review queue could help a content team focus its limited time on pages where review may be most useful, instead of examining every page manually.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*


The decision I want to improve is **which content page should be reviewed first**. The unit of analysis is one anonymized content page, represented by one row in the starter dataset. The planned output is a ranked review queue with a suggested action and simple reason codes for each recommended page.

A content editor or SEO analyst would review the highest-ranked pages and decide whether to refresh, expand, protect, prune, or monitor them. A wrong recommendation could waste limited editor time on a low-value page, while a genuinely important declining page might be missed. Missing a valuable page may be more costly than reviewing one extra page, so I will evaluate the ranking using a top-K metric such as Precision@50 and keep a human reviewer in the final decision.

Data analysis or ML may help because the decision depends on several interacting signals, including impressions, clicks, position, page age, freshness, engagement, and performance movement. However, I will first compare the approach with a transparent rule-based baseline. A more complex model is useful only if it produces a meaningfully better and explainable review queue.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd
from pathlib import Path

# Try repository-relative paths first.
possible_paths = [
    Path("data/raw/content_refresh_anonymized.csv"),
    Path("../../data/raw/content_refresh_anonymized.csv"),
]

data_path = next(
    (path for path in possible_paths if path.exists()),
    None
)

if data_path is not None:
    df = pd.read_csv(data_path)
else:
    # Fallback for a notebook opened directly in Google Colab.
    data_url = (
        "https://raw.githubusercontent.com/"
        "mehkzhra/FlyRank-ML-Internship/main/"
        "data/raw/content_refresh_anonymized.csv"
    )
    df = pd.read_csv(data_url)

# Calculate supporting statistics.
total_pages = len(df)

declining_mask = df["trend_direction"].eq("down")
declining_pages = int(declining_mask.sum())
declining_percentage = declining_mask.mean() * 100

high_exposure_declining_pages = int(
    (
        declining_mask
        & df["impressions_90d"].ge(1000)
    ).sum()
)

print(f"Total anonymized content pages: {total_pages:,}")
print(
    f"Observed declining pages: {declining_pages:,} "
    f"({declining_percentage:.2f}%)"
)
print(
    "Declining pages with at least 1,000 impressions: "
    f"{high_exposure_declining_pages:,}"
)

Total anonymized content pages: 30,000
Observed declining pages: 16,262 (54.21%)
Declining pages with at least 1,000 impressions: 8,031


The starter dataset contains **30,000 anonymized content pages**. Among them, **16,262 pages (54.21%)** are observed as declining. In addition, **8,031 declining pages** received at least 1,000 impressions during the trailing 90-day period.

These numbers show that the content-review problem is large enough to require prioritization. A content team is unlikely to have enough time to review every declining page. Therefore, a ranked queue could help the team focus first on high-exposure pages that may deserve attention. These figures support investigating this lane, but they do not prove that refreshing any selected page will improve its performance.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project will report **observed and measured patterns** in the anonymized dataset. It may identify directional signals associated with declining or high-opportunity pages. The final output will provide **decision support** by ranking which pages deserve human review first. I will describe the selected pages as review candidates rather than guaranteed opportunities.

The analysis cannot prove that refreshing a page causes its traffic, ranking, click-through rate, or engagement to improve because the available data is observational and no controlled experiment has been performed. It also cannot predict or explain Google’s algorithm, guarantee future performance, reconstruct private client information, or prove that a particular feature caused a page to decline. Final decisions and content actions must remain subject to human review.

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit the repo URL on the card